Corzbank.Services/UserService.cs

```diff
@@ -1,15 +1,80 @@
-﻿using Corzbank.Services.Interfaces;
+﻿using AutoMapper;
+using Corzbank.Data.Entities;
+using Corzbank.Data.Entities.Models;
+using Corzbank.Helpers.Validations;
+using Corzbank.Services.Interfaces;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
 using System;
 using System.Collections.Generic;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Corzbank.Services
 {
     public class UserService : IUserService
     {
-        public void Test()
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+        private readonly ValidateUser _validateUser;
+        public UserService(IMapper mapper, UserManager<User> userManager, ValidateUser validateUser)
         {
-            throw new NotImplementedException();
+            _mapper = mapper;
+            _userManager = userManager;
+            _validateUser = validateUser;
+        }
+
+        public async Task<User> GetUserById(Guid id)
+        {
+            return await _userManager.FindByIdAsync(id.ToString());
+        }
+
+        public async Task<IEnumerable<User>> GetUsers()
+        {
+            return await _userManager.Users.ToListAsync();
+        }
+
+        public async Task<User> RegisterUser(UserModel userForRegistration)
+        {
+            if (userForRegistration != null && userForRegistration.Password == userForRegistration.ConfirmPassword)
+            {
+                var user = _mapper.Map<User>(userForRegistration);
+
+                if (_validateUser.UserIsValid(user))
+                {
+                    await _userManager.CreateAsync(user, userForRegistration.Password);
+
+                    return user;
+                }
+            }
+            return null;
+        }
+
+        public async Task<User> UpdateUser(Guid id, UserModel userForUpdate)
+        {
+            if (userForUpdate != null)
+            {
+                var user = GetUserById(id).Result;
```

Comment:
forget about .Result, use await instead

Reply:
resolved
what about controllers?
I'm returning Ok(result.Result) everywhere

Reply:
Resolve it everywhere. [Hint](https://stackoverflow.com/a/32239717)